### PR TITLE
Remove bounds setter in leaf replaced by validator

### PR DESCRIPTION
**Pull request by andresliszt**
## Description

Hello! I talked with @Transurgeon  to take this PR. It removes `bounds` setter on leaf as discussed [here](https://github.com/cvxpy/cvxpy/issues/2599). The setter logic was replaced by a validator and also added a test passing through the validator

Related Issue: #2599

## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [x] Bug fix
- [ ] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [ ] Add our license to new files.
- [x] Check that your code adheres to our coding style.
- [x] Write unittests.
- [x] Run the unittests and check that they’re passing.
- [ ] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

File: cvxpy/tests/test_expressions.py [hunk 1]
```diff
@@ -219,6 +219,74 @@ def test_constants(self) -> None:
...
+        with self.assertRaises(ValueError) as context:
+            x = cp.Variable((2, 2), name="x", bounds=bounds)
+        self.assertEqual(str(context.exception), "-np.inf is not feasible as an upper bound.")
+
+        # Invalid bounds: Lower bound is inf
+        bounds = [np.inf, 10]
+        with self.assertRaises(ValueError) as context:
+            x = cp.Variable((2, 2), name="x", bounds=bounds)
+        self.assertEqual(str(context.exception), "np.inf is not feasible as a lower bound.")
+
```
Comment (Transurgeon) [inline cvxpy/tests/test_expressions.py:289]: can we move this to the file test_attributes.py?

File: cvxpy/expressions/leaf.py [hunk 2]
```diff
@@ -586,6 +580,13 @@ def bounds(self, value):
             elif np.isscalar(val):
                 value[idx] = np.full(self.shape, val)
 
+        # Upper bound cannot be -np.inf.
+        if np.any(value[1] == -np.inf):
+            raise ValueError("-np.inf is not feasible as an upper bound.")
+        # Lower bound cannot be np.inf.
+        if np.any(value[0] == np.inf):
+            raise ValueError("np.inf is not feasible as a lower bound.")
+
```
Comment (Transurgeon) [inline cvxpy/expressions/leaf.py:589]: why did you have to move this?

**Review by Transurgeon — CHANGES_REQUESTED** [review]
generally looks good and seems to be pretty clean with the internal function implementation, we can merge after addressing the comments.
Thanks @andresliszt !

File: cvxpy/expressions/leaf.py (hunk 2 above)
Comment (andresliszt) [reply cvxpy/expressions/leaf.py:589]: because otherwise will fall in the lower bound >= upper bound error

**Review by PTNobel — APPROVED** [review]
Looks good to me once @Transurgeon 's comment is addressed

**Comment by andresliszt** [comment]
I just moved the test location :)

**Comment by andresliszt** [comment]
Oh i messed something in the unit test definition, will be updating this

**Review by Transurgeon — APPROVED** [review]

**Pull request merged**